src/ResourceManager/Search/Commands.Management.Search/Models/PSSearchService.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.Search.Models;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Management.Search.Models
+{
+    public class PSSearchService
+    {
+        public string ResourceGroupName { get; private set; }
+
+        public string Name { get; private set; }
+
+        public string Id { get; private set; }
+
+        public string Location { get; private set; }
+
+        public SkuName? Sku { get; private set; }
+
+        public int? ReplicaCount { get; private set; }
+
+        public int? PartitionCount { get; private set; }
+
+        public HostingMode? HostingMode { get; private set; }
+
+        public IDictionary<string, string> Tags { get; set; }
+
+        public PSSearchService(Azure.Management.Search.Models.SearchService searchService)
+        {
+            ResourceGroupName = ParseResourceGroupFromId(searchService.Id);
+            Name = searchService.Name;
+            Id = searchService.Id;
+            Location = searchService.Location;
+            Sku = searchService.Sku.Name;
+            ReplicaCount = searchService.ReplicaCount;
+            PartitionCount = searchService.PartitionCount;
+            HostingMode = searchService.HostingMode;
+            Tags = searchService.Tags;
+        }
+
+        public static PSSearchService Create(Azure.Management.Search.Models.SearchService searchService)
+        {
+            return new PSSearchService(searchService);
+        }
+
+        private static string ParseResourceGroupFromId(string resouceId)
+        {
+            if (!string.IsNullOrEmpty(resouceId))
```

Comment:
Please use the resource identifier here: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Utilities/Models/ResourceIdentifier.cs